src/main/java/org/codice/itest/api/TestResultFactory.java

```diff
@@ -28,10 +28,11 @@
     /**
      * @param testName - the name of the test that failed.
      * @param exceptionMessage - a description of the failure.
+     * @param stackTrace - the stack trace of the failure
      * @param startTime - the start time of the test
      * @return an appropriate TestResult object.
      */
-    TestResult fail(String testName, String exceptionMessage, Instant startTime, Instant endTime);
```

Comment:
:question: Is there an advantage to treating the stack trace as a string instead of just passing in the `java.lang.Throwable` object?

Reply:
No, I don't think so. At that point a failure and an error would be the same, so would we need both methods?

Reply:
Yes, because they represent different things.  A failure is the result of an `AssertionError`, generally caused by a call to `assertThat()`.  An error is for things like NPEs or some other problem in the setup or execution of the test.  A failure means the test ran but failed, an error means the test failed to run at all because of some misconfiguration.